src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.Factory.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+
+#pragma warning disable CS8601 // Possible null reference assignment.
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+#pragma warning disable 8500 // address / sizeof of managed types
+
+namespace System.Numerics.Tensors
+{
+    public static partial class Tensor
+    {
+        public static Tensor<T> Create<T>(bool mustPin, ReadOnlySpan<nint> lengths)
+            where T : IEquatable<T>
+        {
+            nint linearLength = SpanHelpers.CalculateTotalLength(lengths);
+            T[] values = mustPin ? GC.AllocateArray<T>((int)linearLength, mustPin) : (new T[linearLength]);
```

Comment:
Seems like exposing this mustPin option could result in significant fragmentation of the pinned object heap

Reply:
Would there be a better way to go about this?

@tannergooding

Reply:
Not really. We could hide the API off or give it a more explicit name to better indicate to users the potential danger around it.

The POH is a bit special and the GC really expects it to only be used for extremely long lived allocations (the type you allocate once and let live for the entire lifetime of the app). Pinning such long lived allocations can be important for the user and so we likely want to give users the option so they can get the correct perf/etc, but there may be better alternatives to that.

One example of an alternative would be to allow the user to pass in the `T[]` and indicate we should take ownership of it, rather than doing a defensive copy. This would allow them to pin it themselves if that was appropriate.

Reply:
Should we worry about that now? Or is that a question to solve for after this initial preview?